### ISSUE TITLE
[Bug]: Custom function for `validate_if()` false error report

### ISSUE THREAD

**Issue by alexverse** [Appsilon/data.validator#70]
### Guidelines

- [X] I agree to follow this project's Contributing Guidelines.

### What happened?

Trying to wrap a validation pipeline in a function and reuse it fails. 
It looks like there is some issue with the namespace. We can sidestep this by using the `{{` operator, but is this the intended behavior? 

### Steps to reproduce

``` r
library(dplyr, warn.conflicts = FALSE)
library(data.validator)
library(assertr)

metadata_validator <- function(data_) {
  report <- data_validation_report()
  validate(data_) %>%
    validate_if(nrow(data_) > 0) %>%
    add_results(report)
  report
}

metadata_validator(iris)
#> Validation summary: 
#>  Number of successful validations: 0
#>  Number of failed validations: 1
#>  Number of validations with warnings: 0
#> 
#> Advanced view: 
#> 
#> 
#> |table_name |description |type  | total_violations|
#> |:----------|:-----------|:-----|----------------:|
#> |data_      |NA          |error |                1|
```

<sup>Created on 2023-06-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

There is also the following strange behavior if we define `data_` outside the function.
 
``` r
library(dplyr, warn.conflicts = FALSE)
library(data.validator)
library(assertr)

data_ <- iris

metadata_validator <- function(data_) {
  report <- data_validation_report()
  validate(data_) %>%
    validate_if(nrow(data_) > 0) %>%
    add_results(report)
  report
}

metadata_validator(iris)
#> Validation summary: 
#>  Number of successful validations: 1
#>  Number of failed validations: 0
#>  Number of validations with warnings: 0
#> 
#> Advanced view: 
#> 
#> 
#> |table_name |description |type    | total_violations|
#> |:----------|:-----------|:-------|----------------:|
#> |data_      |NA          |success |               NA|
```

<sup>Created on 2023-06-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

We can sidestep this by using the curly-curly operator but it feels like it is slightly off-label usage:

``` r
library(dplyr, warn.conflicts = FALSE)
library(data.validator)
library(assertr)

metadata_validator <- function(data_) {
  report <- data_validation_report()
  validate(data_) %>%
    validate_if(nrow({{ data_ }}) > 0) %>%
    add_results(report)
  report
}

metadata_validator(iris)
#> Validation summary: 
#>  Number of successful validations: 1
#>  Number of failed validations: 0
#>  Number of validations with warnings: 0
#> 
#> Advanced view: 
#> 
#> 
#> |table_name |description |type    | total_violations|
#> |:----------|:-----------|:-------|----------------:|
#> |data_      |NA          |success |               NA|
```

<sup>Created on 2023-06-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### Expected behavior

Should behave the same with the following,

``` r
library(dplyr, warn.conflicts = FALSE)
library(data.validator)
library(assertr)

data_ <- iris

report <- data_validation_report()
validate(data_) %>%
  validate_if(nrow(data_) > 0) %>%
  add_results(report)

report
#> Validation summary: 
#>  Number of successful validations: 1
#>  Number of failed validations: 0
#>  Number of validations with warnings: 0
#> 
#> Advanced view: 
#> 
#> 
#> |table_name |description |type    | total_violations|
#> |:----------|:-----------|:-------|----------------:|
#> |data_      |NA          |success |               NA|
```

<sup>Created on 2023-06-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
